### PR TITLE
planner: refactor the code of buildUpdate to solve some bugs

### LINKED ISSUE #13201: Wrong update result when stored there's generated column and subquery in UPDATE
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
CREATE TABLE `test_gc_read` (
  `a` int(11) NOT NULL,
  `b` int(11) DEFAULT NULL,
  `c` int(11) GENERATED ALWAYS AS (`a` + `b`) VIRTUAL,
  `d` int(11) GENERATED ALWAYS AS (`a` * `b`) STORED,
  `e` int(11) GENERATED ALWAYS AS (`c` * 2) VIRTUAL,
  PRIMARY KEY (`a`)
);
create table t1(a int);
insert into test_gc_read (a, b) values(3, 4);
insert into t1 values(3);
update test_gc_read set a = a + 100 where a in (select a from t1);
```

2. What did you expect to see?

Both the value of `a`, `c`, `d`, `e` changed.

3. What did you see instead?

The value of `d` is not changed.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### LINKED ISSUE #13202: Error thrown when there're two table with same name but from different database in the `UPDATE` stmt
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
use test;
create table t(a int);
create database test2;
create test2.t(a int);
update test.t, test2.t set test.t.a=test.t.a+100;
```

2. What did you expect to see?

Executes successfully.

3. What did you see instead?

Error `ambiguous column` returned.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

fix #13201 and fix #13202 

### What is changed and how it works?

Before entering `buildUpdateLists`, the passed arg `updateTableList` already used the `AsName` directly generated from ast.
We don't need to get the `AsName` from the created plan.

Also `extractTableList` didn't reset the `DBName` when use the `AsName` to set the `Name`.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

File: executor/update_test.go [hunk 1]
```diff
@@ -225,3 +225,13 @@ func (s *testUpdateSuite) TestUpdateSchemaChange(c *C) {
...
 }
+
+func (s *testUpdateSuite) TestUpdateMultiDatabaseTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop database if exists test2")
+	tk.MustExec("create database test2")
+	tk.MustExec("create table t(a int)")
+	tk.MustExec("create table test2.t(a int)")
+	tk.MustExec("update t, test2.t set test.t.a=1")
```
Comment (lamxTyler) [inline executor/update_test.go]: It does not fail in current master?

Comment (winoros) [reply executor/update_test.go]: fixed

File: executor/update_test.go [hunk 2]
```diff
@@ -225,3 +225,13 @@ func (s *testUpdateSuite) TestUpdateSchemaChange(c *C) {
...
 }
+
+func (s *testUpdateSuite) TestUpdateMultiDatabaseTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop database if exists test2")
+	tk.MustExec("create database test2")
+	tk.MustExec("create table t(a int)")
+	tk.MustExec("create table test2.t(a int)")
+	tk.MustExec("update t, test2.t set test.t.a=1 where test.t.a=2")
```
Comment (lamxTyler) [inline executor/update_test.go]: Still not fail...
```sql
mysql> update t, test2.t set test.t.a=1 where test.t.a=2;
Query OK, 0 rows affected (0.00 sec)
Rows matched: 0  Changed: 0  Warnings: 0

mysql> select tidb_version();
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                                |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-alpha-743-g1b72ce598
Git Commit Hash: 1b72ce5987b3d3396390648cd80788b07b40e4e1
Git Branch: master
UTC Build Time: 2019-11-06 11:44:13
GoVersion: go version go1.13.3 linux/amd64
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

Comment (winoros) [reply executor/update_test.go]: Oh, it needs generated columns too.

**Pull request merged**